include/zephyr/kernel.h

```diff
@@ -14,6 +14,7 @@
 #define ZEPHYR_INCLUDE_KERNEL_H_
 
 #if !defined(_ASMLANGUAGE)
+#include "version.h" /* generated by MAKE, at compile time */
```

Comment:
I'm not fond of this.

`kernel.h` itself doesn't require anything / provide anything related to version, so I fail to see why it should suddenly start to include a header so that other parts of code can save a single include line.

Doing this will have a severe impact on incremental builds.
On incremental builds, then only code that contains actual changes (or headers they include) will rebuild if there is a need.
But including `version.h` in `kernel.h` means that everyone will depend on `version.h`, and thus commiting a `foo.txt`, unrelated to other files will cause everything to rebuild.

Today:
```
$ cmake -GNinja -DBOARD=nrf52840dk_nrf52840 -Bbuild samples/hello_world
$ ninja -C build
$ ninja: Entering directory `build'
...
[134/134] Linking C executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       18152 B         1 MB      1.73%
             RAM:        5580 B       256 KB      2.13%
        IDT_LIST:          0 GB         2 KB      0.00%
$ ninja -C build
ninja: Entering directory `build'
ninja: no work to do.
$ touch foo.txt; git add foo.txt; git commit -s -m"File added"
$ ninja -C build
ninja: Entering directory `build'
[1/9] Generating include/generated/version.h
-- Zephyr version: 3.5.0-rc1 (/projects/github/ncs/zephyr), build: v3.5.0-rc1-45-gb59168b4833f
[9/9] Linking C executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       18152 B         1 MB      1.73%
             RAM:        5580 B       256 KB      2.13%
        IDT_LIST:          0 GB         2 KB      0.00%
```
Note, that the new file causes a new commit, which generates a new `version.h`, but only files actually using `version.h` needs to be recompiled, and thus linked.

Just 9 build step.

If `kernel.h` includes `version.h`, then almost everything will rebuild.
Result of doing the  `#include "version.h"` change in my repo,
```
$ ninja -C build
ninja: Entering directory `build'
ninja: no work to do.
$ touch foo.txt; git add foo.txt; git commit -s -m"File added"
$ ninja -C build
ninja: Entering directory `build'
$ touch foo.txt; git add foo.txt; git commit -s -m"File added"
$ ninja -C build;
ninja: Entering directory `build'
[1/89] Generating include/generated/version.h
-- Zephyr version: 3.5.0-rc1 (/projects/github/ncs/zephyr), build: v3.5.0-rc1-45-gdf07a813c711
[89/89] Linking C executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       18152 B         1 MB      1.73%
             RAM:        5580 B       256 KB      2.13%
        IDT_LIST:          0 GB         2 KB      0.00%
```